--- src-orig/setup.py	2023-05-09 21:03:50.000000000 +0000
+++ src/setup.py	2023-05-09 20:55:00.530036610 +0000
@@ -249,7 +249,7 @@
 # see if the user passed a quiet flag to setup.py arguments and respect
 # that in our parts of the build
 EMIT_BUILD_WARNING = False
-RERUN_CMAKE = False
+RERUN_CMAKE = True
 CMAKE_ONLY = False
 filtered_args = []
 for i, arg in enumerate(sys.argv):
@@ -1021,7 +1021,7 @@
         'nn/parallel/*.pyi',
         'utils/data/*.pyi',
         'utils/data/datapipes/*.pyi',
-        'lib/*.so*',
+        'lib/libtorch_global_deps.so',
         'lib/*.dylib*',
         'lib/*.dll',
         'lib/*.lib',
@@ -1170,6 +1170,12 @@
         'packaged/ATen/native/*',
         'packaged/ATen/templates/*',
     ]
+     # Chaquopy: see package_data below.
+    if os.path.exists("chaquopy"):
+        shutil.rmtree("chaquopy")
+    os.mkdir("chaquopy")
+    os.symlink("../torch/lib", "chaquopy/lib")
+
     setup(
         name=package_name,
         version=version,
@@ -1184,12 +1190,16 @@
         install_requires=install_requires,
         extras_require=extras_require,
         package_data={
+            'chaquopy': [
+                'lib/*.so*'  # See exclude_package_data below.
+            ],
             'torch': torch_package_data,
             'torchgen': torchgen_package_data,
             'caffe2': [
                 'python/serialized_test/data/operator_test/*.zip',
             ],
         },
+        exclude_package_data={'chaquopy': ['lib/libtorch_global_deps.so']},
         url='https://pytorch.org/',
         download_url='https://github.com/pytorch/pytorch/tags',
         author='PyTorch Team',
