--- src-orig/gcc-4.9/gcc/reload.h	2023-05-10 12:31:36.502640669 +0200
+++ src/gcc-4.9/gcc/reload.h	2023-05-10 11:53:30.593423382 +0200
@@ -166,7 +166,7 @@
      value indicates the level of indirect addressing supported, e.g., two
      means that (MEM (MEM (REG n))) is also valid if (REG n) does not get
      a hard register.  */
-  bool x_spill_indirect_levels;
+  int x_spill_indirect_levels;
 
   /* True if caller-save has been reinitialized.  */
   bool x_caller_save_initialized_p;
